pkg/window/strategy/sliding/sliding.go

```diff
@@ -99,8 +105,11 @@ type Windower struct {
 	closedWindows *window.SortedWindowListByEndTime
 }
 
-func NewWindower(length time.Duration, slide time.Duration) window.TimedWindower {
+func NewWindower(length time.Duration, slide time.Duration, vertexInstance *dfv1.VertexInstance) window.TimedWindower {
```

Comment:
Have you thought about adding a new method to the Windower interface `EmitMetrics(vertexInstance *dfv1.VertexInstance)`? That way the windower can emit metrics not only after closing windows but also after other operations, which is up to the caller to decide. It also simplifies the Windower struct itself.

Reply:
I don't think adding Prometheus metrics related functionality to the interface is ideal, the caller shouldn't be tasked with handling these metrics, since they are not directly related to window functionality.